outscale/data_source_outscale_keypair.go

```diff
@@ -16,20 +16,9 @@ import (
 
 func datasourceOutscaleOApiKeyPairRead(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*OutscaleClient).OSCAPI
-	req := oscgo.ReadKeypairsRequest{
-		Filters: &oscgo.FiltersKeypair{KeypairNames: &[]string{d.Id()}},
-	}
-
-	KeyName, KeyNameisOk := d.GetOk("keypair_name")
-	if KeyNameisOk {
-		filter := oscgo.FiltersKeypair{}
-		filter.SetKeypairNames([]string{KeyName.(string)})
-		req.SetFilters(filter)
-	}
-
-	filters, filtersOk := d.GetOk("filter")
+	req := oscgo.ReadKeypairsRequest{}
 
-	if filtersOk {
+	if filters, filtersOk := d.GetOk("filter"); filtersOk {
 		req.SetFilters(buildOutscaleOAPIKeyPairsDataSourceFilters(filters.(*schema.Set)))
 	}
```

Comment:
This is a singular data_source, therefore to avoid matching with multiple data_source user should add
```
filtersilters, filtersOk := d.GetOk("filter")
if !filtersOk {
   return fmt.Errorf("filters must be assigned")
}
```